src/algorithms/base_algorithm.h

```diff
@@ -122,21 +122,14 @@ class BaseAlgorithm {
   // ESTIMATE FUNCTION
   //! Evaluates the logpdf for each single iteration on a given grid of points
   virtual Eigen::MatrixXd eval_lpdf(const Eigen::MatrixXd &grid,
-                                    BaseCollector *const collector) = 0;
-  // TODO will soon become obsolete
-  virtual Eigen::MatrixXd eval_lpdf(const Eigen::MatrixXd &grid,
-                                    const Eigen::MatrixXd &covariates,
+                                    const Eigen::MatrixXd &hier_covariates,
```

Comment:
since this is one of the core 'API' functions, would it be possible to change the signature to

```
virtual Eigen::MatrixXd eval_lpdf(BaseCollector *const collector, const Eigen::MatrixXd &grid, 
                                                     const Eigen::MatrixXd &hier_covariates,
                                                     const Eigen::MatrixXd &mix_covariates)
```
and have default empty values for the last two parameters?